Neo4j.Driver/Neo4j.Driver/V1/Types/CypherDate.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Copyright (c) 2002-2018 "Neo Technology,"
+// Network Engine for Objects in Lund AB [http://neotechnology.com]
+// 
+// This file is part of Neo4j.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using Neo4j.Driver.Internal;
+
+namespace Neo4j.Driver.V1
+{
+    /// <summary>
+    /// Represents a date value, without a time zone and time related components
+    /// </summary>
+    public struct CypherDate : ICypherValue, IEquatable<CypherDate>
+    {
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="CypherDate"/> from individual date component values
+        /// </summary>
+        /// <param name="year"></param>
+        /// <param name="month"></param>
+        /// <param name="day"></param>
+        public CypherDate(int year, int month, int day)
+            : this(new DateTime(year, month, day))
+        {
+
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="CypherDate"/> from a date value
+        /// </summary>
+        /// <param name="date"></param>
+        public CypherDate(DateTime date)
+            : this(date.DaysSinceEpoch())
+        {
+
+        }
+
+        internal CypherDate(long epochDays)
+        {
+            EpochDays = epochDays;
+        }
+
+        /// <summary>
+        /// Days since Unix Epoch
+        /// </summary>
+        public long EpochDays { get; }
```

Comment:
I think it would be nice to expose year, month and day instead. This would make `CypherDate` useful even without additional libraries. Epoch days is kind of an implementation detail.